src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -996,8 +996,8 @@ private static void CheckRestrictedTypeReceiver(BoundExpression expression, Comp
                 {
                     // Because the receiver didn't pass through CoerceArguments, we need to apply an appropriate conversion here.
                     Debug.Assert(argsToParams.IsDefault || argsToParams[0] == 0);
-                    receiverArgument = CreateConversion(receiver, methodResult.Result.ConversionForArg(0), 
-                                                                     GetTypeOrReturnTypeWithAdjustedNullableAnnotations(receiverParameter).TypeSymbol, 
+                    receiverArgument = CreateConversion(receiver, methodResult.Result.ConversionForArg(0),
+                                                                     receiverParameter.Type.TypeSymbol,
```

Comment:
>              [](start = 56, length = 13)

Nit: consider reformatting to have consistent indentation with the previous line. Looks like this is a victim of past renames.